packages/ember-toucan-core/src/components/form/fields/checkbox-group.hbs

```diff
@@ -21,6 +21,10 @@
         {{else}}
           {{@label}}
         {{/if}}
+
+        {{#if this.isReadOnlyOrDisabled}}
+          <this.LockIcon class="text-disabled -ml-1 -mt-0.5 inline-block" />
```

Comment:
I ended up using negative margins on these to kind of knock them into place.

Without the negative top margin, the icon was bottom-aligned with the label:

<img width="218" alt="Screenshot 2023-06-21 at 10 34 11 AM" src="https://github.com/CrowdStrike/ember-toucan-core/assets/8069555/0a0c6cc4-2da0-425d-a5cb-1d5bd39a16cb">

Whereas in the designs its _slightly_ offset like this:

<img width="200" alt="Screenshot 2023-06-21 at 10 34 48 AM" src="https://github.com/CrowdStrike/ember-toucan-core/assets/8069555/ca0c86f3-2aad-4e1c-b083-2f44ef7611c9">

---

The negative margin-left is due to the default spacing of the SVG view box being too large.  Here is without the negative left margin:

<img width="221" alt="Screenshot 2023-06-21 at 10 35 58 AM" src="https://github.com/CrowdStrike/ember-toucan-core/assets/8069555/cc271d73-f630-4b6f-bac4-4e3852ad4d5e">



By using negative-margin-left, it gets closer to the designs.

<img width="218" alt="Screenshot 2023-06-21 at 10 36 14 AM" src="https://github.com/CrowdStrike/ember-toucan-core/assets/8069555/f1c6402b-2899-43d7-bf7a-6a8c85d9125b">

---

I asked Mary and David to review, so we will see what they say

Reply:
Curious—why not the 12-pixel lock with `flex items-center gap-1.5` on the `<legend />`?

Reply:
Good idea. I'll give this a go on Monday and see if it works

Reply:
This feels _wayyyy_ better, thank you for the suggestion! 🙇